src/App.js

```diff
@@ -6,17 +6,25 @@ import "./app.css";
 const today = new Date();
 
 function App() {
-  const [newTodo, setNewTodo] = useState("Hi! Something will be here");
+  //This state change the list of todos;
+  const [todoList, setTodoList] = useState([]);
+
+  //Callback handler, to ad New Todo to List
+  const addTodo = (newTodo) => {
+    console.log("newTodo", newTodo);
+    setTodoList([...todoList, newTodo]);
```

Comment:
You could consider using this **updater function approach**:

`setTodoList((prevState) => [...prevState, newTodo])`

This is considered a more robust and safer way to update state when the new state depends on the previous state. This is especially important when you have multiple state updates happening in the same component, as it ensures that you're working with the most up-to-date state. This approach takes the previous state as a parameter and guarantees that the state is updated correctly.

Instead of:

`setTodoList([...todoList, newTodo])`

This approach might work fine in this specific case since you are just adding an element to the array, and you are not relying on the previous state. However, it's generally not recommended because it can lead to unexpected behavior in more complex scenarios or when multiple state updates are happening in close succession. When using this approach, you're working with the current state todoList, and if there are other state updates happening concurrently, you might not have the latest state.